organizations/views.py

```diff
@@ -104,19 +113,29 @@ def get_object(self, queryset=None):
         Make sure that current user is allowed to delete the shift
         """
         shift = super(ShiftDeleteView, self).get_object()
-
         current_user = self.request.user
-        shift_facility = shift.facility
-        shift_organization = shift.facility.organization
-
-        if (is_manager(current_user, shift_organization, shift_facility)
-            or current_user.is_superuser):
+        facility = shift.facility
+        
+        if (self.can_delete_shift(current_user, facility)):
             return shift
         else:
             raise Http404
 
     def get_success_url(self):
         return reverse("shift_management")
+        
+    def can_delete_shift(self, user, facility):
+        """
+        Check if user has the general deletion rights and belongs
+        to the given facility.
+        """
+        if (not user.has_perm('scheduler.delete_shift')):
+            return False
+        user_facilities = get_cached_memberships(user)[1]
+        for facility_id in user_facilities:
```

Comment:
@yeonwoon  Yes, I have tested this after setting up the demo facility manager like explained on this page: https://github.com/coders4help/volunteer_planner/wiki/How-to-promote-a-user-as-a-%28shift%29-manager-for-facilities.

I am aware that `user_facilities` is a list of integers which I take for the IDs of the facilities a user belongs to. Where do you see a problem with the `for` statement? Did it not work for you? If so, please tell me how to reproduce it.